src/kindhandlers/index.ts

```diff
@@ -139,10 +138,40 @@ export const getDependentResourceKinds = (resourceKinds: string[]) => {
   return [...new Set(dependentResourceKinds)];
 };
 
-export function refMapperMatchesKind(refMapper: RefMapper, kind: string) {
-  if (refMapper.target.kind.startsWith('$')) {
-    return kind.match(refMapper.target.kind.substring(1)) !== null;
-  }
+function readBundledCrdKindHandlers() {
+  const crds = findFiles(getStaticResourcePath(`kindhandlers${path.sep}crds`), '.yaml');
+  crds.forEach(crdPath => {
+    try {
+      const crdContent = fs.readFileSync(crdPath, 'utf-8');
+      if (crdContent) {
+        const documents = parseAllDocuments(crdContent, {prettyErrors: true});
+        documents.forEach(doc => {
+          const crd = doc.toJS({maxAliasCount: -1});
+          if (crd && crd.kind && crd.kind === 'CustomResourceDefinition') {
+            const kindHandler = extractKindHandler(crd, `kindhandlers${path.sep}handlers`);
+            if (kindHandler) {
+              registerKindHandler(kindHandler, false);
+            }
+          }
+        });
+      }
+    } catch (e) {
+      log.warn(`Failed to parse kindhandler CRD at ${crdPath}`, e);
+    }
+  });
+}
 
-  return refMapper.target.kind === kind;
+function findFiles(dir: string, ext: string) {
```

Comment:
Could we rewrite this to be async and use the promise version of `fs` methods?
It would be nice to start doing this for new features so we won't have to refactor more code later if we'll decide to rewrite all I/O functions to be async

Reply:
done - please have a look !